src/imitation/scripts/train_imitation.py

```diff
@@ -3,173 +3,139 @@
+from imitation.scripts.ingredients import bc as bc_ingredient
 from imitation.scripts.ingredients import demonstrations, environment, expert
 from imitation.scripts.ingredients import logging as logging_ingredient
-from imitation.scripts.ingredients import train
+from imitation.scripts.ingredients import policy_evaluation
 
 logger = logging.getLogger(__name__)
 
 
-@train_imitation_ex.capture(prefix="train")
-def make_policy(
-    venv: vec_env.VecEnv,
-    policy_cls: Type[policies.BasePolicy],
-    policy_kwargs: Mapping[str, Any],
-    agent_path: Optional[str],
-) -> policies.BasePolicy:
-    """Makes policy.
-
-    Args:
-        venv: Vectorized environment we will be imitating demos from.
-        policy_cls: Type of a Stable Baselines3 policy architecture.
-            Specify only if policy_path is not specified.
-        policy_kwargs: Keyword arguments for policy constructor.
-            Specify only if policy_path is not specified.
-        agent_path: Path to serialized policy. If provided, then load the
-            policy from this path. Otherwise, make a new policy.
-            Specify only if policy_cls and policy_kwargs are not specified.
-
-    Returns:
-        A Stable Baselines3 policy.
-    """
-    policy_kwargs = dict(policy_kwargs)
-    if issubclass(policy_cls, policies.ActorCriticPolicy):
-        policy_kwargs.update(
-            {
-                "observation_space": venv.observation_space,
-                "action_space": venv.action_space,
-                # parameter mandatory for ActorCriticPolicy, but not used by BC
-                "lr_schedule": utils.get_schedule_fn(1),
-            },
-        )
-    policy: policies.BasePolicy
-    if agent_path is not None:
-        warnings.warn(
-            "When agent_path is specified, policy_cls and policy_kwargs are ignored.",
-            RuntimeWarning,
+def _add_expert_stats_to_stats_dict_if_all_are_trajectories_with_reward(
+    stats: Dict[str, Any],
+    expert_trajs: Sequence[types.Trajectory],
+) -> None:
+    """Adds expert statistics to `stats` if all expert trajectories have reward."""
+    if all(isinstance(t, types.TrajectoryWithRew) for t in expert_trajs):
+        expert_stats = rollout.rollout_stats(
+            cast(Sequence[types.TrajectoryWithRew], expert_trajs),
         )
-        policy = bc_algorithm.reconstruct_policy(agent_path)
+        stats["expert_stats"] = expert_stats
     else:
-        policy = policy_cls(**policy_kwargs)
-    logger.info(f"Policy network summary:\n {policy}")
-    return policy
+        logger.warning(
+            "Expert trajectories do not have reward information, so expert "
+            "statistics cannot be computed.",
+        )
 
 
-@train_imitation_ex.capture
-def train_imitation(
-    bc_kwargs: Mapping[str, Any],
-    bc_train_kwargs: Mapping[str, Any],
-    dagger: Mapping[str, Any],
-    use_dagger: bool,
-    agent_path: Optional[str],
+@train_imitation_ex.command
+def bc(
```

Comment:
Rationale behind splitting them? I guess it does simplify the function signature quite a lot. But I wonder what the original explanation for doing it this way was, and why this is now outweighed.

Reply:
The original decision to have them both in the same function was probably to reduce code duplication? I can only guess here. BC is needed as an ingredient to DAgger so there is some kind of hierarchical dependency between the two.

I think it is conceptually cleaner to split them. In the long term I want to have each algorithm in its own ingredient and its own script and this is a first step in that direction. For more details you can read the discussion starting [here](https://github.com/HumanCompatibleAI/imitation/issues/79#issuecomment-1401904262).